File: RecommendationEngine/RecommendationEngine/Models/DB/Asset.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace RecommendationEngine.Models
{
    public class Asset
    {
        private int assetId { get; set; }
        private string name { get; set; }
        private string displayText { get; set; }
        private string energyType { get; set; } 
        private string timeZone { get; set; } 
        private string elementPath { get; set; } 
        private double acPower { get; set; }
        private Asset assetChild { get; set; }
    }
}

File: RecommendationEngine/RecommendationEngine/FrontEnd/src/entities/Asset.ts
//You have to make sure that the method that fetches this object
//have exactly this attribute, or it will break and it will be hard to debug.
export interface Asset {
    Id: number,
    Name: string,
    DisplayText: string,
    EnergyType: string,
    TimeZone: string,
    ElementPath: string,
    AcPower: string,
    Children: Asset[]
}
